src/main/java/de/tum/in/www1/artemis/domain/ProgrammingExerciseBuildConfig.java

```diff
@@ -0,0 +1,200 @@
+package de.tum.in.www1.artemis.domain;
+
+import java.util.Objects;
+
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.OneToOne;
+import jakarta.persistence.Table;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.JsonProcessingException;
+
+import de.tum.in.www1.artemis.service.connectors.aeolus.Windfile;
+import de.tum.in.www1.artemis.service.connectors.vcs.AbstractVersionControlService;
+
+@Entity
+@Table(name = "programming_exercise_build_config")
+@JsonInclude(JsonInclude.Include.NON_EMPTY)
+public class ProgrammingExerciseBuildConfig extends DomainObject {
+
+    private static final Logger log = LoggerFactory.getLogger(ProgrammingExerciseBuildConfig.class);
+
+    @Column(name = "sequential_test_runs")
+    private Boolean sequentialTestRuns;
+
+    @Column(name = "branch")
+    private String branch;
+
+    @Column(name = "build_plan_configuration", columnDefinition = "longtext")
+    private String buildPlanConfiguration;
+
+    @Column(name = "build_script", columnDefinition = "longtext")
+    private String buildScript;
+
+    /**
+     * This boolean flag determines whether the solution repository should be checked out during the build (additional to the student's submission).
+     * This is currently only supported for HASKELL and OCAML, thus the default value is false.
+     */
+    @Column(name = "checkout_solution_repository", columnDefinition = "boolean default false")
+    private boolean checkoutSolutionRepository;
+
+    @Column(name = "checkout_path")
+    private String checkoutPath;
+
+    @Column(name = "timeout_seconds")
+    private Integer timeoutSeconds;
```

Comment:
Do we have a default value here? Would be reasonable in my opinion.
If so, please use the base type.
```suggestion
    private int timeoutSeconds;
```